OMGVA PoS/Business layer/Controllers/BusinessController.cs

```diff
@@ -0,0 +1,86 @@
+﻿using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Newtonsoft.Json;
+using OMGVA_PoS.Data_layer.DTOs;
+using OMGVA_PoS.Data_layer.Enums;
+using OMGVA_PoS.Data_layer.Models;
+using OMGVA_PoS.Data_layer.Repositories.Business_Management;
+using OMGVA_PoS.Helper_modules.Utilities;
+using System.IdentityModel.Tokens.Jwt;
+
+namespace OMGVA_PoS.Business_layer.Controllers
+{
+    [Route("business")]
+    [ApiController]
+    public class BusinessController(IBusinessRepository businessRepository) : Controller
+    {
+        private readonly IBusinessRepository _businessRepository = businessRepository;
+
+        [HttpGet]
+        [Authorize(Roles = "Admin")]
+        [ProducesResponseType<List<Business>>(StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status401Unauthorized)]
+        [ProducesResponseType(StatusCodes.Status403Forbidden)]
+        public IActionResult GetAllBusinesses() {
+            return Ok(JsonConvert.SerializeObject(_businessRepository.GetBusinesses()));
+        }
+
+
+        [HttpGet("{id}")]
+        [Authorize(Roles = "Admin,Owner")]
+        [ProducesResponseType<Business>(StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status401Unauthorized)]
+        [ProducesResponseType(StatusCodes.Status403Forbidden)]
+        [ProducesResponseType(StatusCodes.Status404NotFound)]
+        public IActionResult GetBusiness(long id) {
+            JwtSecurityToken token = JwtTokenHelper.GetJwtToken(HttpContext.Request.Headers.Authorization);
+            if (token.UserRoleEquals(UserRole.Owner) && !token.UserBusinessEquals(id)) {
+                return Forbid();
+            }
+            Business business = _businessRepository.GetBusiness(id);
+
+            if (business == null)
+                return NotFound();
+            else
+                return Ok(JsonConvert.SerializeObject(business));
+        }
+
+        [HttpPost]
+        [Authorize(Roles = "Admin")]
+        [ProducesResponseType<Business>(StatusCodes.Status201Created)]
+        [ProducesResponseType(StatusCodes.Status400BadRequest)]
+        [ProducesResponseType(StatusCodes.Status401Unauthorized)]
+        [ProducesResponseType(StatusCodes.Status403Forbidden)]
+        public IActionResult CreateBusiness([FromBody]CreateBusinessRequest createBusinessRequest) {
+            if (!createBusinessRequest.Email.IsValidEmail() || !createBusinessRequest.Phone.IsValidPhone()){
+                return BadRequest();
+            }
```

Comment:
Minor, but just a thought. Wouldn't it be better that `StringFormatValidator.cs` would `throw` an error instead of returning `true`/`false` ? That way the error would be more specific, since we could specify whether the phone number or email was invalid in the request